src/linux/link.rs

```diff
@@ -100,15 +100,23 @@ impl SharedObject {
             if let Ok(rel) = rpath.strip_prefix(encoded_prefix) {
                 let new_rpath = prefix.join(rel);
                 let relative_path = pathdiff::diff_paths(
-                    new_rpath,
+                    &new_rpath,
                     self.path.parent().ok_or(RelinkError::NoParentDir)?,
                 )
                 .ok_or(RelinkError::PathDiffFailed)?;
-                tracing::info!("New relative path: $ORIGIN/{}", relative_path.display());
-                final_rpath.push(PathBuf::from(format!(
-                    "$ORIGIN/{}",
-                    relative_path.to_string_lossy()
-                )));
+                if relative_path.starts_with("..") {
```

Comment:
Does `Component::Parent` work with `starts_with`, too? For Path's.

Reply:
Yes,

```rust
use std::path::Path;

fn main()  {
    let path = Path::new("../test");
    assert!(path.starts_with(".."));
    assert_eq!(Some(Path::new("..")), path.parent());
}
```

Do you prefer checking that instead?